### DISCUSSION TITLE
SPI on crazyflie

### DISCUSSION THREAD

**Discussion by SriramKodey** [bitcraze/discussions#1519] [Q&A]
Hi all,
I want to set up SPI between the crazyflie (master) and an STM32F407 disc board (slave). Do you think I can use deck_spi.c API file for this? Is there any example code I can use as a guide?
Thanks

**Comment by ataffanel** [comment]
Yes, you can use the deck_spi API. You can find a couple of deck drivers using this API, for example the flow sensor driver is using it: https://github.com/bitcraze/crazyflie-firmware/blob/ac565120e6ca2858d2e4a73f204775502e2d7233/src/drivers/src/pmw3901.c#L181

**Reply by SriramKodey** [reply]
Is there any documentation for the deck_spi API?

**Reply by ataffanel** [reply]
No there is no documentation, the API was designed to look like the Arduino API and there is often some description in the header file for each functions: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/deck/interface/deck_spi.h

If you have any specific question or problem do not hesitate to ask.
